### PR TITLE
feat(apollo): Apollo studio integration

### PR DESCRIPTION
**Pull request by shashitnak**
**Summary:**  
Changes for integration Apollo Studio with tailcall

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
    - Introduced Apollo tracing for enhanced server-side telemetry in GraphQL.
    - Added support for Apollo settings in telemetry configuration.
    - Implemented schema extensions for async GraphQL, allowing for dynamic extension registration.
- **Bug Fixes**
    - Streamlined initialization processes for `AppContext` and `ServerConfig` with proper asynchronous handling.
- **Chores**
    - Updated CI workflows to ensure Protoc is available, removing conditional checks for smoother operation.
    - Added a Python script to set the `PROTOC` environment variable, improving build and test setup.
- **Documentation**
    - Added examples and tests showcasing Apollo Tracing functionality.

### CODE REVIEW

File: .github/scripts/python-script.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import os
+import shutil
+
+print("Running python script...")
+os.environ["PROTOC"] = shutil.which("protoc")
+print("{}".format(os.environ["PROTOC"]))
```
Comment (tusharmath) [inline .github/scripts/python-script.py:6]: needs to be deleted before merging.

File: aws-lambda/src/main.rs [hunk 2]
```diff
@@ -39,11 +39,8 @@ async fn main() -> Result<(), Error> {
         .await?;
     let blueprint = Blueprint::try_from(&config)?;
 
-    let app_ctx = Arc::new(AppContext::new(
-        blueprint,
-        runtime,
-        config.extensions.endpoints,
-    ));
+    let app_ctx = AppContext::new(blueprint, runtime, config.extensions.endpoints).await?;
```
Comment (tusharmath) [inline aws-lambda/src/main.rs:42]: Avoid performing side-effect while creating instances of structs. Let's add new method called `register_apollo_schema` which is an async method, that can be called on AppContext if required.

File: src/config/apollo.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
+use serde::{Deserialize, Serialize};
+
+use crate::config::ConfigReaderContext;
+use crate::mustache::Mustache;
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq, schemars::JsonSchema)]
+pub struct Apollo {
```
Comment (tusharmath) [inline src/config/apollo.rs:7]: Use camelcase for all

File: Cargo.toml [hunk 4]
```diff
@@ -124,6 +124,14 @@ opentelemetry-prometheus = "0.15.0"
 phonenumber = "0.3.3"
 chrono = "0.4.35"
 
+# TODO: move this flag into `cli`, like other dependencies.
```
Comment (tusharmath) [inline Cargo.toml:127]: drop the comment.

File: Cargo.toml [hunk 5]
```diff
@@ -124,6 +124,14 @@ opentelemetry-prometheus = "0.15.0"
...
+async-graphql-extension-apollo-tracing = { git = "https://github.com/shashitnak/async_graphql_apollo_studio_extension/", branch = "std-protobuf" }
```
Comment (tusharmath) [inline Cargo.toml:128]: Move this repo into `tailcallhq`

File: Cargo.toml [hunk 6]
```diff
@@ -124,6 +124,14 @@ opentelemetry-prometheus = "0.15.0"
...
+
+[patch.crates-io]
+# TODO: update this once https://github.com/haixuanTao/opentelemetry-system-metrics/pull/2 gets merged
+async-graphql-value = {git ="https://github.com/tusharmath/async-graphql.git", branch = "add-setter"}
+async-graphql = {git ="https://github.com/tusharmath/async-graphql.git", branch = "add-setter"}
```
Comment (tusharmath) [inline Cargo.toml:133]: Move this repo into `tailcallhq`

File: src/config/apollo.rs (hunk 3 above)
Comment (shashitnak) [reply src/config/apollo.rs:8]: fixed

File: Cargo.toml (hunk 4 above)
Comment (shashitnak) [reply Cargo.toml]: fixed

File: aws-lambda/src/main.rs (hunk 2 above)
Comment (shashitnak) [reply aws-lambda/src/main.rs]: removed `register` call since it's not working properly

File: Cargo.toml (hunk 5 above)
Comment (shashitnak) [reply Cargo.toml]: moved

**Pull request merged**